MapboxNavigation/Array.swift

```diff
@@ -32,3 +35,22 @@ extension Array where Element: NSAttributedString {
         return joinedAttributedString
     }
 }
+
+extension Array where Element == RouteStep {
+    // Find the longest contiguous series of RouteSteps connected to the first one.
+    //
+    // tolerance: -- Maximum distance between the end of one RouteStep and the start of the next to still consider them connected. Defaults to 100 meters
+    func continuousShape(tolerance: CLLocationDistance = 100) -> LineString? {
+        guard count > 0 else { return nil }
+        guard count > 1 else { return self[0].shape }
+        var continuousLine = [CLLocationCoordinate2D]()
+
+        for index in 0...count-2 {
+            if let currentStepFinalCoordinate = self[index].shape?.coordinates.last, currentStepFinalCoordinate.distance(to: self[index+1].maneuverLocation) < tolerance, let coordinates = self[index].shape?.coordinates {
```

Comment:
Manual index arithmetic makes me nervous. Can we express this code differently using Sequence methods like `prefix(…)` and `zip(…)`?

```suggestion
        for index in 0...count-2 {
            if let coordinates = self[index].shape?.coordinates, let currentStepFinalCoordinate = coordinates.last, currentStepFinalCoordinate.distance(to: self[index+1].maneuverLocation) < tolerance {
```

then, since `maneuverLocation` is guaranteed to be the first coordinate in `shape`:

```swift
zip(compactMap { $0.shape }, suffix(from: 1).compactMap { $0.shape })
    .filter { $0.coordinates.last?.distance(to: $1.coordinates.first ?? Double.max) < tolerance }
```

Reply:
@1ec5 - I believe I have incorporated your feedback in the change I just made. Let me know if I missed an improvement to this function that you had suggested.